src/libs/auth/RedirectFromOAuth.ts

```diff
@@ -0,0 +1,22 @@
+import { UserManager } from 'oidc-client-ts';
+import { Spinner } from '../../components/Spinner';
+import { OidcBroker } from './oidcBroker';
+
+const RedirectFromOAuth = (): JSX.Element => {
+  const userManager: UserManager = new UserManager(
+    OidcBroker.getUserManagerSettings()
+  );
+  const url = window.location.href;
+  const isSilent = window.location.pathname.startsWith(
+    '/redirect-from-oauth-silent'
+  );
+
+  if (isSilent) {
+    userManager.signinSilentCallback(url);
+  } else {
+    userManager.signinPopupCallback(url);
+  }
```

Comment:
This looks like it belongs in a `useEffect` hook instead of in `render`.

But does this need to be in a React component at all? It looks like `oauth-redirect-loader` could call the `UserManager` functions directly (and render a Spinner if desired).

Reply:
I don't think this needs to be in a React component since this should only happen once each time the user signs in through the popup window, and then I believe that app instance ends once that window closes.